packages/next/src/server/app-render/app-render.tsx

```diff
@@ -1848,13 +1855,8 @@ async function prerenderToStream(
             // will track it again there
             dynamicTracking: null,
             revalidate: INFINITE_CACHE,
-            tags: null,
+            tags: [...ctx.requestStore.implicitTags],
```

Comment:
why do we have to duplicate the implicit tags in both it's own set and in the tags set?

Reply:
The implicitTags is an immutable set that is used for reads of any inner caches (but they're not used to tag those caches).

The outer ISR page also gets explicitly tagged with the implicit tags because it doesn't have a separate concept of soft tags.

This set is mutable and will have other things added to it which inner caches shouldn't see.